src/orchestrator.rs

```diff
@@ -92,37 +111,34 @@ impl Orchestrator {
     pub async fn on_start_up(&self, health_check: bool) -> Result<(), Error> {
         info!("Performing start-up actions for orchestrator...");
         if health_check {
-            info!("Probing health status of configured clients...");
-            // Run probe, update cache
-            let res = self.clients_health(true).await.unwrap_or_else(|e| {
-                // Panic for unexpected behaviour as there are currently no errors propagated to here.
-                panic!("Unexpected error during client health probing: {}", e);
-            });
+            info!("Probing client health...");
+            let client_health = self.client_health(true).await;
```

Comment:
should this be donw in a tokio task so that it doesn't block startup ?  (we can address it in future, but just wanted to make a note here)

Reply:
I assume `Orchestrator::on_start_up()` tasks, i.e. checks, should "complete" prior to startup, but I don't really have an opinion on this. @pscoro any thoughts on this?

Reply:
On one side, its good to check if everything is wired up correctly, but then in case connections and this verification takes time, it will halt orchestrator bootup and look like orchestrator is having issue.

So I think we should make this async and put out `error` levels logs if this startup check fails for some downstreams.

(can be handled in separate PR though)

Reply:
I do recall that we wanted to leverage health checks as an extension of config validation, so that we should catch misconfigured services at start-up.

Reply:
yep, thats accurate, but then we realized that this has a unwanted side-effect that it will make orchestrator's startup to fail when even 1 of the downstream is not responding.  And so it looks like orchestrator is having issue (from looking at deployment), but the real problem is one of the detectors. 

So the idea pivoted to orchestrator doing this check, but instead of blocking its own startup / boot, it will expose this information with API call.

Reply:
> I assume `Orchestrator::on_start_up()` tasks, i.e. checks, should "complete" prior to startup, but I don't really have an opinion on this. @pscoro any thoughts on this?

This brings up a good point, the naming of that function may have some legacy from when we used to want orchestrator to fatally error if the start up health check failed. Now since we decided that the orchestrator does not treat its clients as dependencies.

I think `on_startup()` should imply behaviour executed before start up, maybe in the future this could include some tasks that need to run synchronously. I think what we should do is run the call to `client_health` in a new tokio task thats started inside the `on_start_up()`. Thoughts on this?

Reply:
> I do recall that we wanted to leverage health checks as an extension of config validation, so that we catch misconfigured services at start-up. I believe this was the original intention of running probes at start-up.

This would still catch misconfig errors at start up time. Since we are already decided that the state of clients (as they are known via their configs) does not effect the liveliness of the orchestrator, and are only reported as non-fatal errors, I dont think there is functionally any difference between:
- starting an async health check, then starting the orchestrator & recving requests, then handling and reporting the response of the health check without crashing on failure, and 
- starting a sync health check, waiting for the response, then handling and reporting the response of the health check without crashing on failure, and then starting the orchestrator